components/kyma-environment-broker/internal/process/provisioning/ems_bind.go

```diff
@@ -0,0 +1,201 @@
+package provisioning
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/google/uuid"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/ptr"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/runtime/components"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/servicemanager"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/control-plane/components/provisioner/pkg/gqlschema"
+	"github.com/sirupsen/logrus"
+)
+
+type EventingOverrides struct {
+	OauthClientId      string `json:"oauthClientId"`
+	OauthClientSecret  string `json:"oauthClientSecret"`
+	OauthTokenEndpoint string `json:"oauthTokenEndpoint"`
+	PublishUrl         string `json:"publishUrl"`
+	BebNamespace       string `json:"bebNamespace"`
+}
+
+type EmsBindStep struct {
+	operationManager *process.ProvisionOperationManager
+	secretKey        string
+}
+
+func NewEmsBindStep(os storage.Operations, secretKey string) *EmsBindStep {
+	return &EmsBindStep{
+		operationManager: process.NewProvisionOperationManager(os),
+		secretKey:        secretKey,
+	}
+}
+
+var _ Step = (*EmsBindStep)(nil)
+
+func (s *EmsBindStep) Name() string {
+	return "EMS_Bind"
+}
+
+func (s *EmsBindStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	if !operation.Ems.Instance.ProvisioningTriggered {
+		return s.handleError(operation, fmt.Errorf("Step %s : Ems Provisioning step was not triggered", s.Name()), log, "")
+	}
+
+	smCli, err := operation.ServiceManagerClient(log)
+	if err != nil {
+		return s.handleError(operation, err, log, fmt.Sprintf("Step %s : unable to create Service Manage client", s.Name()))
+	}
+	// test if the provisioning is finished, if not, retry after 10s
+	resp, err := smCli.LastInstanceOperation(operation.Ems.Instance.InstanceKey(), "")
+	if err != nil {
+		return s.handleError(operation, err, log, fmt.Sprintf("Step %s : LastInstanceOperation() call failed", s.Name()))
+	}
+	log.Infof("Step %s : Provisioning Ems (instanceID=%s) state: %s", s.Name(), operation.Ems.Instance.InstanceID, resp.State)
```

Comment:
the `log` instance contains data like "step name" so you don't need to put it in the message.

Reply:
The code on line 61 generates the following log message:
```
  time="2020-12-21T17:20:29+01:00" level=info msg="Step EMS_Bind : Provisioning Ems (instanceID=ff9a4806-abca-41ea-94e3-2ac7a139e38a) state: in progress"
```
Without having "Step %s:..." at the beginning of the message and "s.Name()" as a variable, no step name was logged.
This log extract was taken from running of integration test.

I've removed the "step name" from the message, cause I saw that the "manager" will set it before executing the step on production:
```
		for _, step := range steps {
			logStep := logOperation.WithField("step", step.Name())
			logStep.Infof("Start step")

			processedOperation, when, err = m.runStep(step, processedOperation, logStep)
```
 The superfluous "step name" was removed from all log messages.

Reply:
Done